x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LogicalPlanOptimizer.java

```diff
@@ -142,11 +142,11 @@ protected static Batch<LogicalPlan> substitutions() {
             new ReplaceAggregateNestedExpressionWithEval(),
             new ReplaceRegexMatch(),
             new ReplaceTrivialTypeConversions(),
+            new ReplaceOrderByExpressionWithEval(),
+            new PropagateInlineEvals(),
```

Comment:
1. Let's add a comment to mention that this has to come before `ReplaceAliasingEvalWithProject` so that it can move the `EVAL` generated by `ReplaceAggregateNestedExpressionWithEval` into the left child of `InlineJoin`s.
2. Why don't we place this immediately after `ReplaceAggregateNestedExpressionWithEval`? These are the dependent rules - `ReplaceRegexMatch` and `ReplaceTrivialTypeConversions` shouldn't matter between these two, no?

Reply:
I just tried the latter and the single node EsqlSpecIT is still passing.